woocommerce/woocommerce-gutenberg-products-block
# Replace phpcs rule

## What

This PR replaces the removed `WordPress.PHP.DisallowShortTernary` in favor of `Universal.Operators.DisallowShortTernary`.

Ref: https://github.com/WordPress/WordPress-Coding-Standards/blob/7722c0b8a6f4eb615be4ba8ebe22ed92a5fa87d7/CHANGELOG.md?plain=1#L231

## Why

Since the rule is removed, it is causing an error such as this in the VS Code editor ![file.png](https://github.com/woocommerce/woocommerce-blocks/assets/2132595/01a615be-bc18-4b04-8089-f7835dc82bab)


## Note

I am not sure why we're trying to exclude the `src/*` [folder](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/phpcs.xml#L38) since this is the folder where the bulk of the PHP files lives. I would think this is exactly the place we should lint? Perhaps you know cc @nielslange 

~~Also this is blocked by WooCommerce Sniffs need to be first updated to version `1.0.0` so we can use WP coding standards version `^3.0.0`.~~

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Have PHPCS installed globally.
2. Install the WooCommerce Sniffs `composer global require woocommerce/woocommerce-sniffs --dev` (version 1.0.0)
3. Add the PHPCS extension to your VS Code.
4. Change the execution path of the PHPCS extension to point to your globally installed PHPCS (use `which phpcs` to find the path).
5. Set the PHPCS extension standard to `WooCommerce-Core` in the `settings.json` like `"phpcs.standard": "WooCommerce-Core"`.
6. Open up a PHP file outside of `/src/*` and try to add a shorthand ternary operation and ensure PHPCS catches that. You can use `$foobar = true ?: false;`.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> N/A
